core/arch/arm/kernel/generic_boot.c

```diff
@@ -836,23 +839,42 @@ static void update_fdt(void)
 		panic();
 	}
 }
+#endif /*CFG_DT && !CFG_EMBEDDED_SECURE_DT*/
 
-#else
-static void init_fdt(unsigned long phys_fdt __unused)
+#if defined(CFG_DT) && defined(CFG_EMBEDDED_SECURE_DT)
+void *get_dt_blob(void)
 {
+	assert(cpu_mmu_enabled());
+
+	if (!dt_blob_addr) {
+		if (fdt_check_header((void *)&static_secure_dtb[0]))
```

Comment:
Cast is not needed
`&static_secure_dtb[0]` seems like a complicated way of writing `static_secure_dtb` so why not use the latter?
Same comment at the `dt_blob_addr` assignment a few lines below.

Reply:
true